server/docs/getting-started.md

```diff
@@ -2,106 +2,251 @@
+1. Use the `AppInstance`'s `takeAction()` method to [update state](https://specs.counterfactual.com/en/latest/01-app-definition.html#progressing-state) in the virtual channel
+1. Use the `AppInstance`'s `uninstall()` method to close and resolve the virtual channel
 
-in three ways:
+Along the way, we'll see how the Counterfactual framework ensures that state is updated safely and securely. 
 
-1. **Starting a channel** - When users request / agree to play a game, the UI passes the request (via connection 1) to the Counterfactual node. The node then instantiates a state channel based on the game logic described by the contract (connection 2). Any staked funds the state channel requires are drawn from the Counterfactual wallet inside MetaMask.
+One more thing: to streamline your first dapp, we've built and deployed a bot that will accept `proposeInstallVirtual()` requests for **HighRoller** and play the game with your user. This means you only have to code the UI for the first player of **HighRoller**.
 
-2. **State management in a channel** - In an open channel, requests to modify state are passed to the CF node (via connection 1). The node calls the pure functions of the solidity contract (via connection 2) to verify that requests to modify state are valid, and to alter the state accordingly. The node makes updated state available to members of the channel (via connection 1).
 
-3. **Ending a game**  - When the game is over, the dapp (HighRoller.js) must request (via connection 1) that the node end the game. In turn, the CF node will (via connection 2) verify that the game is over, and if it is over, have the contract implement the transactions that resolve the game.
 
+---------
 
-### In this Getting Start Guide, you’ll learn how to:
+## A Quick Look at the Contract
+
+Let's take a quick look at the [HighRoller.sol](https://github.com/counterfactual/monorepo/blob/master/packages/apps/contracts/HighRollerApp.sol) contract to see how the game logic is structured.
+
+### Game logic for HighRoller.sol
+
+We want to make our dice game very secure: the application logic requires both users to submit a random number -- the product of the two numbers is used as the source of randomness for both players' dice rolls; this is what makes it secure. In order to prevent the second player from using information about the first player's number, we use a commit-by-hash and reveal paradigm:
+  * The first player submits a **hash** of their **number** with a random **salt**
+  * The second player submits their **number**
+  * The first player **reveals** their number by submitting the number and the salt, which are checked against the original hash
+  * The contract generates die rolls for each player
+
+The information above is enough to determine the structure of the **HighRoller** `AppState` and `Action`, both of which are required for Counterfactual Apps. We also note the data structures `ActionType` and `Stage` that the contract developer chose to create:
```

Comment:
> that the contract developer chose to create:

I said this in my review but it is awkward to literally write this sentence in the guide itself. Basically we want to have some kind of enum for unique types of actions and "stages" of the game but these are just enums and are used _inside_ the `AppState` and `Action` structures to make the `applyAction` function easier to read.